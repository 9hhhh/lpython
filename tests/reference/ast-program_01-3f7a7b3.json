{
    "basename": "ast-program_01-3f7a7b3",
    "cmd": "lfortran --show-ast --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/program_01.f90",
    "infile_hash": "917ed02637c7a6c4e5e5407a0ccf45b4ef6ff79355c5c080e53d97e9",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast-program_01-3f7a7b3.stdout",
    "stdout_hash": "faaf9cace29111d91c333f9d2a1a651adacfbe5c625eb40cc7e66132",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}