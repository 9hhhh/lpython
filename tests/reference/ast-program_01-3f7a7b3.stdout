(TranslationUnit [(Program program_01 [] [(Declaration [(dp "integer" [] [(Attribute parameter [] [])] (FuncCallOrArray kind [(Real "0.d0")] []))]) (Declaration [(a "real" [] [] ())]) (Declaration [(i "integer" [] [] ())])] [(Print () [(Str "Normal random numbers:")]) (DoLoop i 1 10 () [(SubroutineCall rand [a]) (Print () [a])])] [(Subroutine rand [(x)] [] [(Declaration [(x "real" [] [(Attribute intent [(out)] [])] ())]) (Declaration [(first "logical" [] [(Attribute save [] [])] (Constant Unimplementedconstant))]) (Declaration [(u "real" [(1 2)] [(Attribute save [] [])] ())]) (Declaration [(r2 "real" [] [] ())])] [(If first [(DoLoop () () () () [(SubroutineCall random_number [u]) (= u (- (* 2 u) 1)) (= r2 (FuncCallOrArray sum [(** u 2)] [])) (If (BoolOp (< r2 1) And (> r2 0)) [(Exit)] [])]) (= u (* u (FuncCallOrArray sqrt [(/ (* (u- 2) (FuncCallOrArray log [r2] [])) r2)] []))) (= x (FuncCallOrArray u [1] []))] [(= x (FuncCallOrArray u [2] []))]) (= first (not first))] [])])])
